x/evm/statedb/statedb.go

```diff
@@ -526,3 +535,22 @@ func (s *StateDB) Commit() error {
 	}
 	return nil
 }
+
+func (s *StateDB) convertNativeEvents(events []sdk.Event, contract common.Address) {
+	if len(s.eventConverters) == 0 {
+		return
+	}
+
+	if len(events) == 0 {
+		return
+	}
+
+	for _, event := range events {
+		if converter, ok := s.eventConverters[event.Type]; ok {
+			for _, log := range converter(event.Attributes) {
+				log.Address = contract
```

Comment:
Do we update BlockNumber here as well?

Reply:
I'm doing another PR to handle block number and block hash.